src/screens/HomeScreen.tsx

```diff
@@ -1,48 +1,34 @@
-import { useEffect, useState } from 'react';
+import { useEffect } from 'react';
 
 import { ContainerApp } from 'components/containerApp/ContainerApp';
 import { ListProducts } from 'components/listProducts/ListProducts';
-import { Loader } from 'components/loader/Loader';
-import { NoFound } from 'components/noFound/NoFound';
 import { SelectCategoryHome } from 'components/selectCategoryHome/SelectCategoryHome';
-import { CardProduct } from 'model/Product';
 import { materialService } from 'services/materialService';
 import { productService } from 'services/productService';
 import useCategoryStore from 'store/CategoryStore';
 import useMaterialStore from 'store/MaterialStore';
+import { useFetchQuery } from 'utils/hooks/useFetchQuery';
+import { useParamsQuery } from 'utils/hooks/useParamsQuery';
 
 export const HomeScreen = () => {
-    const [products, setProducts] = useState<CardProduct[]>([]);
-    const [loading, setLoading] = useState<boolean>(false);
-
     const setMaterials = useMaterialStore((state) => state.setMaterials);
     const category = useCategoryStore((state) => state.category);
 
-    useEffect(() => {
-        const fetchData = async () => {
-            setLoading(true);
-
-            const dataProducts = await productService.getAllProductCards(category);
-            const dataMaterials = await materialService.getAllProductMaterials();
-            setProducts(dataProducts.pageData);
-            setMaterials(dataMaterials);
+    const { data: products, isLoading } = useFetchQuery(
+        productService.getAllProductCards,
+        useParamsQuery({ path: category })
+    );
 
-            setLoading(false);
-        };
+    const { data: dataMaterials } = useFetchQuery(materialService.getAllProductMaterials);
 
-        fetchData();
-    }, [category]);
+    useEffect(() => {
+        dataMaterials && setMaterials(dataMaterials);
+    }, [dataMaterials]);
 
     return (
         <>
-            <ContainerApp>
-                {loading ? (
-                    <Loader />
-                ) : products.length > 0 ? (
-                    <ListProducts products={products} />
-                ) : (
-                    <NoFound />
-                )}
+            <ContainerApp isData={!!products && products.pageData.length > 0} isLoading={isLoading}>
```

Comment:
`!!products && products.pageData.length > 0` replace to `!!products?.pageData?.length`